app/src/main/java/com/imdbmovieapp/data/repository/FavoriteMovieRepositoryImpl.kt

```diff
@@ -0,0 +1,31 @@
+package com.imdbmovieapp.data.repository
+
+import com.imdbmovieapp.data.local.database.FavoriteMovieDatabase
+import com.imdbmovieapp.data.local.mapper.MovieDomainToMovieEntityMapper
+import com.imdbmovieapp.data.local.mapper.MovieEntityToMovieDomainMapper
+import com.imdbmovieapp.domain.model.FavoriteMovieDomain
+import com.imdbmovieapp.domain.repository.FavoriteMovieRepository
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+class FavoriteMovieRepositoryImpl(
+    private val favoriteMovieDatabase: FavoriteMovieDatabase,
+    private val movieEntityToMovieDomainMapper: MovieEntityToMovieDomainMapper,
+    private val movieDomainToMovieEntityMapper: MovieDomainToMovieEntityMapper
+) : FavoriteMovieRepository {
+    override suspend fun insert(favoriteMovieDomain: FavoriteMovieDomain) {
+        return favoriteMovieDatabase.movieDao()
+            .insertMovie(movieDomainToMovieEntityMapper.mapModel(favoriteMovieDomain))
```

Comment:
favoriteMovieDatabase.movieDao().insertMovie(
            movieDomainToMovieEntityMapper.mapModel(favoriteMovieDomain)
        )
This style of formatting looks better for me